standalone-metastore/metastore-server/src/test/java/org/apache/hadoop/hive/metastore/client/TestGetFileList.java

```diff
@@ -0,0 +1,237 @@
+  }
+
+  @After
+  public void tearDown() throws Exception {
+    try {
+        client.close();
+    } finally {
+      client = null;
+    }
+  }
+
+  protected AbstractMetaStoreService getMetaStore() {
+    return metaStore;
+  }
+
+  protected IMetaStoreClient getClient() {
+    return client;
+  }
+
+  protected void setClient(IMetaStoreClient client) {
+    this.client = client;
+  }
+
+  @Test
+  public void testGetFileListResultCountOnePartition() throws Exception {
+    createDatabaseOnePartition();
+    List<String> partVals = new ArrayList<>();
+    partVals.add("a1");
+    GetFileListResponse resp = getClient().getFileList(null, DB_NAME, TABLE_NAME, partVals,  "");
+    Assert.assertEquals(NUM_FILES_IN_PARTITION, resp.getFileListDataSize());
+  }
+
+  @Test
+  public void testGetFileListResultOnePartition() throws Exception {
+    createDatabaseOnePartition();
+    List<String> partVals = new ArrayList<>();
+    partVals.add("a1");
+    GetFileListResponse resp = getClient().getFileList(null, DB_NAME, TABLE_NAME, partVals,  "");
+    List<String> expectedPaths = new ArrayList<>();
+    for (int i = 0; i < NUM_FILES_IN_PARTITION; i++) {
+      expectedPaths.add("/" + FILE_NAME + i);
+    }
+
+    List<String> actualPaths = new ArrayList<>();
+    for (ByteBuffer buffer: resp.getFileListData()) {
+      FbFileStatus fileStatus = FbFileStatus.getRootAsFbFileStatus(buffer);
+      actualPaths.add(fileStatus.relativePath());
+    }
+    Collections.sort(actualPaths);
+    Assert.assertArrayEquals(expectedPaths.toArray(), actualPaths.toArray());
+  }
+
+  @Test
+  public void testGetFileListResultMultiPartition() throws Exception {
+    createDatabaseMultiPartition();
+    List<String> partVals = new ArrayList<>();
+    partVals.add("a1");
+    partVals.add("1");
+    GetFileListResponse resp = getClient().getFileList(null, DB_NAME, TABLE_NAME, partVals,  "");
+    List<String> expectedPaths = new ArrayList<>();
+    for (int i = 0; i < NUM_FILES_IN_PARTITION; i++) {
+      expectedPaths.add("/" + FILE_NAME + i);
+    }
+
+    List<String> actualPaths = new ArrayList<>();
+    for (ByteBuffer buffer: resp.getFileListData()) {
+      FbFileStatus fileStatus = FbFileStatus.getRootAsFbFileStatus(buffer);
+      actualPaths.add(fileStatus.relativePath());
+    }
+    Collections.sort(actualPaths);
+    Assert.assertArrayEquals(expectedPaths.toArray(), actualPaths.toArray());
+  }
+
+  @Test
+  public void testGetFileListNoResult() throws Exception {
+    createDatabaseOnePartition();
+    List<String> partVals = new ArrayList<>();
+    partVals.add("a2");
+    Assert.assertThrows(MetaException.class,
+            () -> getClient().getFileList(null, DB_NAME, TABLE_NAME, partVals,  ""));
+  }
+
+
+
+  private void createDatabaseOnePartition() throws Exception {
+    String path = hiveWarehouse;
+
+    Database db = new DatabaseBuilder().
+            setName(DB_NAME).
+            setLocation(path).
```

Comment:
Why do you need to set the location?
Weren't default location working for this test?